### ISSUE TITLE
IdentifierName claims java.lang.String is mutable

### ISSUE THREAD

**Issue by ben-manes** [google/error-prone#5523]
```java
Context.java:92: warning: [IdentifierName] Static variables should be named in UPPER_SNAKE_CASE if deeply immutable or lowerCamelCase if not, with acronyms treated as words (https://google.github.io/styleguide/javaguide.html#s5.3-camel-case), but ACTION_ClASS_COMPONENT_TYPE is not
    private static final String ACTION_ClASS_COMPONENT_TYPE = "actionClass";
                                ^
    (see https://google.github.io/styleguide/javaguide.html#s5.2-specific-identifier-names)
  Did you mean 'private static final String aCtionClAsscomponenttype = "actionClass";'?
```

```java
@AutoFactory
@DoNotMock("..."
public final class Context {
  ...
  public static final class TaggedRunnable implements CheckedRunnable {
    private static final String ACTION_ClASS_COMPONENT_TYPE = "actionClass";
    ...
  }
}

**Comment by ben-manes** [comment]
oh, there was a lower case L, `ClASS`. The warning message didn't point out the typo by assuming it was mutable.
